### PR TITLE
Retry time fixes

### PR DESCRIPTION
**Pull request by karknu**
# Description

Fix problems related to big ledger and public root peers retry time.

# Checklist

### Quality
* [x] Commit sequence makes sense and have useful messages, see [ref][contrib#git-history].
* [ ] New tests are added and existing tests are updated.
* [x] Self-reviewed the PR.

### Maintenance
* [x] Linked an [issue][link-issue] or added the PR to the current sprint of [`ouroboros-network`][project] project.
* [x] Added labels.
* [x] Updated changelog files.
* [ ] The documentation has been properly updated, see [ref][contrib#documentation].

[project]: https://github.com/orgs/IntersectMBO/projects/5/views/1
[link-issue]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=
[contrib#git-history]: https://github.com/IntersectMBO/ouroboros-network/blob/master/CONTRIBUTING.md#git-history
[contrib#documentation]: https://github.com/IntersectMBO/ouroboros-network/blob/master/CONTRIBUTING.md#documentation

### CODE REVIEW

**Comment by bolt12** [comment]
These all look sensible to me :)

File: ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/BigLedgerPeers.hs [hunk 1]
```diff
@@ -106,9 +106,9 @@ jobReqBigLedgerPeers PeerSelectionActions{ requestPublicRootPeers }
                                bigLedgerPeerBackoffs'
                                bigLedgerPeerRetryDiffTime'],
             decisionState = st {
-                              inProgressPublicRootsReq = False,
-                              publicRootBackoffs  = bigLedgerPeerBackoffs',
-                              publicRootRetryTime = bigLedgerPeerRetryTime'
+                              inProgressBigLedgerPeersReq = False,
+                              bigLedgerPeerBackoffs       = bigLedgerPeerBackoffs',
+                              bigLedgerPeerRetryTime      = bigLedgerPeerRetryTime'
```
Comment (bolt12) [inline ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/BigLedgerPeers.hs:111]: Nice catch!

**Review by bolt12 — APPROVED** [review]

**Review by coot — APPROVED** [review]
Very nice.

**Review by bolt12 — APPROVED** [review]

File: ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/PeerSelection.hs [hunk 2]
```diff
@@ -686,11 +686,21 @@ tooBusyForTooLong trace0 =
...
         busySpanLength = diffTime busyEndTime busyStartTime
 
+        -- If the governor wakes up due to an action that gives us new credits
+        -- we take those credits into account before failing.
+        endCredits !c [] = c
+        endCredits !c ((t, _, MockEnvEvent e) : tr) | t == busyEndTime =
+          endCredits (c + fromIntegral (envEventCredits e)) tr
+        endCredits !c ((t, _, _) : tr) | t == busyEndTime =
+          endCredits c tr
+        endCredits !c _ = c
```
Comment (coot) [inline ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/PeerSelection.hs:702]: :+1:

**Comment by coot** [comment]
One of the tests failed on Hydra
```
        node never uses non-trustable peers in clean state:                     FAIL (2.74s)
          *** Failed! Falsified (after 51 tests and 77 shrinks):
          GovernorMockEnvironment {peerGraph = PeerGraph [(PeerAddr 10,[],GovernorScripts {peerShareScript = Script (Nothing :| []), peerSharingScript = Script (PeerSharingDisabled :| []), connectionScript = Script ((Noop,NoDelay) :| [])})], localRootPeers = fromGroups [(HotValency {getHotValency = 1},WarmValency {getWarmValency = 1},fromList [(PeerAddr 10,(DoAdvertisePeer,IsTrustable))])], publicRootPeers = PublicRootPeers {getPublicConfigPeers = fromList [], getBootstrapPeers = fromList [], getLedgerPeers = fromList [], getBigLedgerPeers = fromList []}, targets = Script ((PeerSelectionTargets {targetNumberOfRootPeers = 0, targetNumberOfKnownPeers = 1, targetNumberOfEstablishedPeers = 0, targetNumberOfActivePeers = 0, targetNumberOfKnownBigLedgerPeers = 0, targetNumberOfEstablishedBigLedgerPeers = 0, targetNumberOfActiveBigLedgerPeers = 0},NoDelay) :| [(PeerSelectionTargets {targetNumberOfRootPeers = 0, targetNumberOfKnownPeers = 0, targetNumberOfEstablishedPeers = 0, targetNumberOfActivePeers = 0, targetNumberOfKnownBigLedgerPeers = 0, targetNumberOfEstablishedBigLedgerPeers = 0, targetNumberOfActiveBigLedgerPeers = 0},NoDelay)]), pickKnownPeersForPeerShare = Script (PickFirst :| []), pickColdPeersToPromote = Script (PickFirst :| []), pickWarmPeersToPromote = Script (PickFirst :| []), pickHotPeersToDemote = Script (PickFirst :| []), pickWarmPeersToDemote = Script (PickFirst :| []), pickColdPeersToForget = Script (PickFirst :| []), peerSharingFlag = PeerSharingDisabled, useBootstrapPeers = Script ((UseBootstrapPeers [RelayAccessDomain "test5" 48,RelayAccessDomain "test5" 65511,RelayAccessAddress "0.246.32.126" 44,RelayAccessDomain "test1" 65494,RelayAccessAddress "12d:21f:56:5bfe:1e1:bef7:c3:da85" 2,RelayAccessAddress "1e4:3395:1a8:1c05:145:be84:124:520c" 65523,RelayAccessAddress "1.82.19.128" 65531,RelayAccessDomain "test5" 48,RelayAccessDomain "test3" 65500,RelayAccessDomain "test5" 65525,RelayAccessAddress "e6:c8c5:119:9f6c:12d:3aa2:88:beaf" 22,RelayAccessDomain "test3" 65525,RelayAccessAddress "0.100.68.30" 65519,RelayAccessDomain "test2" 50,RelayAccessAddress "0.121.48.177" 65497,RelayAccessAddress "0.40.44.218" 16,RelayAccessDomain "test3" 39,RelayAccessDomain "test5" 65492,RelayAccessDomain "test3" 65512,RelayAccessAddress "0.79.102.117" 65533,RelayAccessDomain "test4" 35,RelayAccessAddress "1bc:1a5a:15d:7af1:19d:a8fc:fd:a338" 40,RelayAccessDomain "test5" 65510,RelayAccessAddress "1.118.216.216" 41,RelayAccessDomain "test4" 6,RelayAccessDomain "test4" 16,RelayAccessAddress "ad:49ba:1ed:fcb5:72:5dc6:147:be7f" 9,RelayAccessDomain "test2" 65491,RelayAccessDomain "test3" 65504,RelayAccessAddress "1ce:7887:1f5:8717:155:e24d:1e6:d5cb" 45,RelayAccessDomain "test1" 65497,RelayAccessAddress "0.7.117.105" 42,RelayAccessAddress "5f:ac20:11a:ff14:130:3e5e:69:3dcd" 65506,RelayAccessDomain "test4" 65500,RelayAccessDomain "test1" 65505,RelayAccessAddress "21:c6b1:29:ac56:36:60d2:ca:2eda" 11,RelayAccessDomain "test3" 65487,RelayAccessDomain "test5" 11],NoDelay) :| []), useLedgerPeers = Script ((DontUseLedgerPeers,NoDelay) :| []), ledgerStateJudgement = Script ((TooOld,NoDelay) :| [])}
          Last 20 signal values:
          Time 0s       : (False,(fromList [],fromList []))
          Time 0s       : (True,(fromList [],fromList []))
          Time 0s       : (True,(fromList [PeerAddr 10],fromList [PeerAddr 10]))
          
          Property violated at: Time 0s
          Invalid signal value:
          (True,(fromList [PeerAddr 10],fromList []))
          
          Use --quickcheck-replay=833314 to reproduce.
          Use -p '/node never uses non-trustable peers in clean state/' to rerun this test only.
```

**Comment by karknu** [comment]
> -p '/node never uses non-trustable peers in clean state/'

This breakage seems to have been introduced in `a38e5a1b63` , that is it exist in master.

**Comment by bolt12** [comment]
I have looked into it and it seems it happens on an edge case when there's a trusted local root peer being in progress promote to cold and then the targets change to be all 0 (due to the quickcheck arbitrary script). This will clamp the local roots to the `targetKnownPeer` limit which will effectively make them empty. However the peer will stay in the `knownPeer` set. In other scenario the peer would have been shortly forgotten but since it is in the `inProgressPromoteCold` set it will linger and make the test fail

**Pull request merged**